src/flow.js

```
@@ -1274,6 +1309,16 @@
             return;
         }
       }
+      if (typeof read === 'function') {
```

right. given that the read is always necessary (not like the process one that cna be set as null) we can simply remove the if. this way users that will set it wrong will notice an error that instead would be silently masked. given that currently flow.js does not validayte any opt i';m ok for removing the if.